server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -353,11 +363,63 @@ public short getDefaultReplication(Path path) {
     return getFileSystemByPath(path).getDefaultReplication(path);
   }
 
+  /**
+   * The Hadoop Configuration object does not currently allow for duplicate properties to be set in
+   * a single Configuration for different FileSystem URIs. Here we will look for properties in the
+   * Accumulo configuration of the form:
+   *
+   * <pre>
+   * instance.volume.config.&lt;volume-uri&gt;.&lt;hdfs-property&gt;
+   * </pre>
+   *
+   * We will use these properties to return a new Configuration object that can be used with the
+   * FileSystem URI.
+   *
+   * @param conf AccumuloConfiguration object
+   * @param hadoopConf Hadoop Configuration object
+   * @param filesystemURI Volume Filesystem URI
+   * @return Hadoop Configuration with custom overrides for this FileSystem
+   */
+  private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
+      final Configuration hadoopConf, final String filesystemURI) {
+    final Configuration volumeConfig = new Configuration(hadoopConf);
+    final Map<String,String> customProps =
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUMES_CONFIG);
+    customProps.forEach((key, value) -> {
+      if (key.startsWith(filesystemURI)) {
+        String property = key.substring(filesystemURI.length() + 1);
+        log.info("Overriding property {} to {} for volume {}", property, value, filesystemURI);
+        volumeConfig.set(property, value);
+      }
+    });
+    return volumeConfig;
+  }
+
+  private static void warnVolumeOverridesMissingVolume(AccumuloConfiguration conf,
+      Set<String> definedVolumes) {
+    final Map<String,String> overrideProperties = new ConcurrentHashMap<>(
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUMES_CONFIG));
+
+    definedVolumes.forEach(vol -> {
+      log.debug("Looking for defined volume: {}", vol);
+      overrideProperties.keySet().forEach(override -> {
+        if (override.startsWith(vol)) {
+          log.debug("Found volume {}, removing property {}", vol, override);
+          overrideProperties.remove(override);
+        }
+      });
+    });
+
+    overrideProperties.forEach((k, v) -> log
+        .warn("Found no matching volume for volume config override property {} = {}", k, v));
+  }
```

Comment:
On second thought, the entire method can be written without an intermediate map or dealing with any concurrency... just grab the properties you want to log a warning about.

```suggestion
  private static void warnVolumeOverridesMissingVolume(AccumuloConfiguration conf,
      Set<String> definedVolumes) {
    conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUMES_CONFIG).entrySet().stream()
        // log only configs where none of the volumes (with a dot) prefix its key
        .filter(e -> definedVolumes.stream().noneMatch(vol -> e.getKey().startsWith(vol + ".")))
        .forEach(
            e -> log.warn("Found no matching volume for volume config override property {}", e));
  }
```

Reply:
Addressed in f34eb45.